content/swiftui/concepts/views/terms/button/button.md

```diff
@@ -1,52 +1,93 @@
+
+```pseudo
+Button(action: { // Initiates when the button is pressed }) {
+  // A Label closure or view
+}
+```
+When creating a button, the `.label` property can be left out and its details can be provided in a separate closure after the button call. This provides more flexibility for customizing the button's appearance.
 
 ## Example
 
-The following example increases the `count` each time the `Button` is used:
+The following example demonstrates the use of a `Button`. Pressing the button will randomly change the colors of the screen and the button itself:
 
 ```swift
-@State var count = 0
-var body: some View {
-  Button("Click me: \(count)") {
-    self.count += 1
-  }
-  .font(.title)
-  .padding()
-  .backgroundColor(Color.green)
-  .cornerRadius(40)
-  .foregroundColor(Color.black)
-  .overlay(RoundedRectangle(cornerRadius: 40).stroke(Color.green))
+struct ContentView: View {
+    @State var backgroundColor = Color(red: 0.69, green: 0.96, blue: 1.0)
+    @State var buttonColor = Color(red: 0.82, green: 0.34, blue: 0.45)
+    
+    var body: some View {
+        ZStack {
+            backgroundColor.ignoresSafeArea()
+            
+            VStack {
+                Button(
+                    action: { randomizeColors() }, // Action to randomize the colors when the button is pressed
+                    label: {
+                        Text("Press This") // The label text of the button
+                            .font(.title)
+                            .fontWeight(.semibold)
+                            .foregroundColor(.white)
+                            .padding()
+                            .padding(.horizontal, 20)
+                            .background(
+                                buttonColor
+                                    .cornerRadius(30)
+                                    .shadow(radius: 5)
+                            )
+                    }
+                )
+            }
+        }
+    }
+    
+    func randomizeColors() -> Void {
+        backgroundColor = Color(
+            red: Double.random(in: 0...1),
+            green: Double.random(in: 0...1),
+            blue: Double.random(in: 0...1)
+        )
+        buttonColor = Color(
+            red: Double.random(in: 0...1),
+            green: Double.random(in: 0...1),
+            blue: Double.random(in: 0...1) 
+        )
+    }
+}
 ```
 
-Each time the user selects the `Button`, the number next to the text `"Click Me"` will increase `self.count` by 1.
+- Pressing the button triggers the `.randomizeColors()` method, which generates random RGB values for the `.backgroundColor` and `.buttonColor` properties. 
+
+The following will be displayed:
+
+![swiftui-button-medium](https://github.com/kevveth/docs/assets/68872850/3f6283f5-bef2-452f-a635-a7608e115426 "Gif of a SwiftUI button being pressed. The colors of the screen and button change randomly as a result.")
```

Comment:
This is not how we host images in Docs. From the [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#file-hosting):

- If you'd like to host files such as images, or GIFs, please upload it to the [/media folder](https://github.com/Codecademy/docs/tree/main/media).
- Images should be linked to using the following url: "https://raw.githubusercontent.com/Codecademy/docs/main/media/" followed by the image name.
- File names should use dashes (`-`) to separate words rather than spaces or underscores.
- File names should be in lower case.
[...]
> **Note:** Image links will not work until a PR is merged.

Reply:
Also, please rename the image something human-readable.